### ISSUE TITLE
Use `type="number | tel"` for <TimePicker /> inputs in order for users to see numeric keyboard on mobile devices

### ISSUE THREAD

**Issue by yuyokk** [palantir/blueprint#3434]
Hi folks,

Right now `<TimePicker />` component renders the following inputs

e.g.

```jsx
<input class="bp3-timepicker-input bp3-timepicker-hour" value="8">
```

so users see the following keyboard on mobile devices 

<img width="585" alt="timepicker" src="https://user-images.githubusercontent.com/1225343/54723026-7dc01b00-4b66-11e9-9cf7-dbb328105950.png">

And to type time they need to do extra clicks on their keyboard (to switch to numeric).

I was wondering if it is possible to change input type to either `number` or `tel` in order for mobile users to see numeric keyboard by default.

Please let me know if this request makes sense.
Thank you 🙏

**Comment by yuyokk** [comment]
Team, I want to work on this one. Just wanted to clarify the solution before starting the work.

Will adding [`type="tel"` to <input />](https://github.com/palantir/blueprint/blob/develop/packages/datetime/src/timePicker.tsx#L219) here be a good enough solution? Or you have any other views on this?

Please let me know.

**Comment by adidahiya** [comment]
This isn't a telephone number, so you should use `type="number"`. But you need to make sure this doesn't interfere with desktop usage, since Blueprint is a desktop-focused UI toolkit.

**Comment by connorlurring** [comment]
I saw this was tagged as a good first issue, so I took a look at this today.

The noticeable difference between the current implementation and using `type-"number"` are the input spinners that the number field provides. These spinners can be disabled using CSS through webkit/moz however.

An alternative that could be used for this field is the HTML5 `type="time"` option, which would degrade into `type="text"` where not supported.

**Comment by saruman1234** [comment]
There's also the [`inputmode` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode). 

```
<input type="text" inputmode="numeric" />
```

Avoids the [accessibility & odd problems of `<input type="number"/>`](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/) but ensures mobile users are shown the numeric keyboard.

- https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode
- https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/

**Comment by neerajbachani** [comment]
using type="tel" is a reasonable solution, but testing on different devices is recommended to ensure consistent behavior. If necessary, you can also try type="number" as an alternative.

**Comment by dev-ramij** [comment]
Team, If the issue is not resolved I will work on this issue

**Comment by evansjohnson** [comment]
It looks like this was addressed by https://github.com/palantir/blueprint/pull/6568 with adding `type=number` - I am not seeing this reproduce.

There was a [comment](https://github.com/palantir/blueprint/pull/6568#discussion_r1408380839) mentioning not finding the reasons for not originally using `type=number` - the items mentioned in the [GOV.UK article in the comment above](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/) seem to either be fixed or not apply.

---

I'll open a separate issue to track investigating adding `pattern="[0-9]*"` to this component since only offering the actual numbers on mobile instead of the full number keyboard with math symbols may be a better experience and only numbers are useful here.
